app/Http/Controllers/ProfileController.php

```diff
@@ -8,58 +8,57 @@
 
 class ProfileController extends Controller
 {
+    public function index()
+    {
+        return view('userList', ['profile_users' => User::paginate(12), 'user' => Auth::user()]);
+    }
+
     public function profile(User $profile_user)
     {
-        $count_words_learned = $profile_user->countWordsLearned();
-        $relation = Auth::user()->relation($profile_user);
+        if ($profile_user->id == Auth::id()) {
+            return redirect ('/dashboard');
```

Comment:
you can use Middleware of laravel to filter users is Authenticated User.
https://laravel.com/docs/5.7/middleware

Reply:
This is not really authentication. This makes that when user try to access to profile page of himself, the dashboard page is appeared instead. Authentication func is made in next after next PR. (Which authenticate admin and logged in user.)

Reply:
middleware is not only for authentication, it can do other things than authentication. you can use it to identify if it is admin or normal user and many more. Middleware are filters not authenticator only

Reply:
I understand. Let me fix.